test/integration/framework/executor/framework.go

```diff
@@ -0,0 +1,133 @@
+}
+
+// ExecOptions passed to ExecWithOptions
+type ExecOptions struct {
+	Client        kubernetes.Interface
+	Command       []string
+	Namespace     string
+	PodName       string
+	ContainerName string
+	Stdin         io.Reader
+	CaptureStdout bool
+	CaptureStderr bool
+	// If false, whitespace in std{err,out} will be removed.
+	PreserveWhitespace bool
+}
+
+type defaultExecutor struct {
+	client kubernetes.Interface
+}
+
+// NewExecutor creates a new instance of Executor for a specific
+// Kubernetes client.
+func NewExecutor(client kubernetes.Interface) Executor {
+	return &defaultExecutor{client: client}
+}
+
+// ExecWithOptions executes a command in the specified container,
+// returning stdout, stderr and error. `options` allowed for
+// additional parameters to be passed.
+func ExecWithOptions(ctx context.Context, options ExecOptions) (stdout, stderr string, err error) {
+	const tty = false
+	req := options.Client.Kubernetes().CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(options.PodName).
+		Namespace(options.Namespace).
+		SubResource("exec").
+		Param("container", options.ContainerName).
+		Context(ctx)
+	req.VersionedParams(&v1.PodExecOptions{
+		Container: options.ContainerName,
+		Command:   options.Command,
+		Stdin:     options.Stdin != nil,
+		Stdout:    options.CaptureStdout,
+		Stderr:    options.CaptureStderr,
+		TTY:       tty,
+	}, scheme.ParameterCodec)
+
+	var stdoutBuff, stderrBuff bytes.Buffer
+	err = execute("POST", req.URL(), options.Client.RESTConfig(), options.Stdin, &stdoutBuff, &stderrBuff, tty)
+
+	if options.PreserveWhitespace {
+		return stdoutBuff.String(), stderrBuff.String(), err
+	}
+	return strings.TrimSpace(stdoutBuff.String()), strings.TrimSpace(stderrBuff.String()), err
+}
+
+// ExecCommandInContainerWithFullOutput executes a command in the
+// specified container and return stdout, stderr and error
+func (e *defaultExecutor) ExecCommandInContainerWithFullOutput(ctx context.Context, namespace, podName, containerName string, cmd ...string) (stdout, stderr string, err error) {
+	return ExecWithOptions(ctx, ExecOptions{
+		Client:             e.client,
+		Command:            cmd,
+		Namespace:          namespace,
+		PodName:            podName,
+		ContainerName:      containerName,
+		Stdin:              nil,
+		CaptureStdout:      true,
+		CaptureStderr:      true,
+		PreserveWhitespace: false,
+	})
+}
+
+// ExecCommandInContainer executes a command in the specified container.
+// Command is expect to succeed.
+func (e *defaultExecutor) ExecCommandInContainer(ctx context.Context, namespace, podName, containerName string, cmd ...string) (stdout string) {
+	stdout, stderr, err := e.ExecCommandInContainerWithFullOutput(ctx, namespace, podName, containerName, cmd...)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred(), `failed to execute command %in pod "%s/%s", container %s: %s`, podName, containerName, stderr)
```

Comment:
You could push the expect to the test and keep the executor clean from test imports/expectations.